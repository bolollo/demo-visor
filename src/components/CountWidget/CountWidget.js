// @flow

import React from "react";
import PropTypes from "prop-types";

import styles from "./CountWidget.module.css";

export default function CountWidget({label, count, position, locale}) {

	const positionClassName = styles[position];

	return (<div className={[styles.countWidget, positionClassName].join(" ")}>
		<div className={styles.labelStyle}>{label}</div>
		<div className={styles.countStyle}>{count.toLocaleString(locale)}</div>
	</div>);

}

CountWidget.propTypes = {
	label: PropTypes.string,
	count: PropTypes.number,
	position: PropTypes.oneOf(["bottom-right", "bottom-left", "top-right", "top-left", "relative"]),
	locale: PropTypes.string
};

CountWidget.defaultProps = {
	label: "Count",
	count: 0,
	position: "top-right",
	locale: "es-ES"
};
