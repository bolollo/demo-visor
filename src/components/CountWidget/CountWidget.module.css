.countWidget {
    z-index:100;
    margin: 0.5rem;
    padding: 5px;
    background-color: #ffffff;
    border-radius: 5px;
    min-width: 170px;
    opacity: 0.9;
}

.bottom-right{
    position: absolute;
    bottom: 0;
    right: 0;
}
.bottom-left{
    position: absolute;
    bottom: 0;
    left: 0;
}
.top-right{
    position: absolute;
    top: 0;
    right: 0;
}
.top-left{
    position: absolute;
    top: 0;
    left: 0;
}
.relative{
    position: relative;
}

.labelStyle {
    font-size: 1em;
    color: #6c757d;
}

.countStyle {
    font-size: 2em;
    color: #343a40;
    text-align: center;
}