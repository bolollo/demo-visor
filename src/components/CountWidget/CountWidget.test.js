// @flow

import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import CountWidget from "./CountWidget";

describe("CountWidget component", () => {

	it("matches the snapshot ", () => {

		const component = shallow(<CountWidget></CountWidget>);
		expect(toJSON(component)).toMatchSnapshot();

	});

	it("has count equal 3", () => {

		const component = shallow(<CountWidget count={3}></CountWidget>);
		expect(component.find(".countStyle").text()).toEqual("3");

	});

	it("has count locale en-US text to be 3,000,000", () => {

		const component = shallow(<CountWidget count={3000000} locale={"en-US"}></CountWidget>);
		expect(component.find(".countStyle").text()).toEqual("3,000,000");

	});

	it("has label equal Total", () => {

		const component = shallow(<CountWidget label={"Total"}></CountWidget>);
		expect(component.find(".labelStyle").text()).toEqual("Total");

	});

	it("has position bottom-left", () => {

		const component = shallow(<CountWidget position={"bottom-left"}></CountWidget>);
		expect(component.find(".countWidget").hasClass("bottom-left")).toEqual(true);

	});

});
