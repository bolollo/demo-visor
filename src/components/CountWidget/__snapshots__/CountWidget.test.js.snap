// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountWidget component matches the snapshot  1`] = `
<div
  className="countWidget top-right"
>
  <div
    className="labelStyle"
  >
    Count
  </div>
  <div
    className="countStyle"
  >
    0
  </div>
</div>
`;
